src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce<GroupsMap<T>>((acc, item) => {
+    const keyValue = String(item[key]);
+
+    if (!acc[keyValue]) {
+      acc[keyValue] = [];
```

Comment:
You can use optional chaining and nullish coalescing to simplify this line. Replace this block with: `acc[keyValue] = acc[keyValue] ?? [];`